lib/graphql/schema/warden.rb

```diff
@@ -146,10 +147,29 @@ def union_memberships(obj_type)
         @unions[obj_type]
       end
 
-      def visible_field?(field_defn)
+      def visible_field?(field_defn, type_defn = nil)
+        return false if type_defn && !field_on_visible_interface?(field_defn, type_defn)
         visible?(field_defn) && visible_type?(field_defn.type.unwrap)
       end
 
+      # Returns true when the field is on a visible interface
+      # returns true if the field is not on an interface
+      # returns false if the field is on a hidden interface
+      def field_on_visible_interface?(field_defn, type_defn)
```

Comment:
This function is a bit funky so I will explain what I am doing:

Check if the field is on an interface that is attached to an object it is visible on.

Ie: If interface x has field y
object z implements interface x

If interface x is hidden, show field y, if it is visible, hide field x.

I need all three of these objects in order to determine a fields visibility so I grab the owner from the field to get the interface.